codenvy-ide-core/src/test/java/com/codenvy/ide/part/editor/EditorPartStackPresenterTest.java

```diff
@@ -0,0 +1,355 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package com.codenvy.ide.part.editor;
+
+import com.codenvy.ide.Resources;
+import com.codenvy.ide.api.editor.EditorPartPresenter;
+import com.codenvy.ide.api.parts.*;
+import com.codenvy.ide.texteditor.openedfiles.ListOpenedFilesPresenter;
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.dom.client.Node;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.event.logical.shared.CloseHandler;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.AcceptsOneWidget;
+import com.google.web.bindery.event.shared.EventBus;
+import com.googlecode.gwt.test.GwtModule;
+import com.googlecode.gwt.test.GwtTestWithMockito;
+import com.googlecode.gwt.test.utils.GwtReflectionUtils;
+import junit.framework.TestCase;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.*;
+import org.vectomatic.dom.svg.OMSVGSVGElement;
+import org.vectomatic.dom.svg.ui.SVGImage;
+import org.vectomatic.dom.svg.ui.SVGResource;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Mockito.*;
+
+/**
+ * Testing {@link com.codenvy.ide.part.editor.EditorPartStackPresenter} functionality.
+ *
+ * @author Alexander Andrienko
+ */
+@GwtModule("com.codenvy.ide.Core")
+public class EditorPartStackPresenterTest extends GwtTestWithMockito {
+
+    @Mock
+    private PartStackUIResources partStackUIResources;
+
+    @Mock
+    private Resources resources;
+
+    @Mock
+    private EditorPartStackView view;
+
+    @Mock
+    private EditorPartPresenter part1;
+
+    @Mock
+    private EditorPartPresenter part2;
+
+    @Mock
+    private EditorPartPresenter part3;
+
+    @Mock
+    private PartStackView.TabItem item;
+
+    @Mock
+    private PartStackView.TabItem item2;
+
+    @Mock
+    private PartStackView.TabItem item3;
+
+    @Mock
+    private EventBus eventBus;
+
+    @Mock
+    private ListOpenedFilesPresenter listOpenedFilesPresenter;
+
+    @Mock
+    private AsyncCallback asyncCallback;
+
+    @Mock
+    private List<PartPresenter> parts;
+
+    @Mock
+    private  PartStackView.ActionDelegate actionDelegate;
+
+    @Mock
+    private EditorPartStackPresenter.PartStackEventHandler partStackEventHandler;
+
+    private EditorPartStackPresenter presenter;
+
+    @Before
+    public void init(){
+        MockitoAnnotations.initMocks(this);
+        presenter = new EditorPartStackPresenter(view, eventBus, partStackEventHandler, listOpenedFilesPresenter);
```

Comment:
It's doesn't work. When I use @InjectMocks, mockito in some test moment, changes one mocks to another and test crashed (It's very strange...). When I use handle instance of presenter all works fine.